src/rules.c

```diff
@@ -126,6 +126,18 @@ SchemaRule *SchemaRule_Create(SchemaRuleArgs *args, StrongRef ref, QueryError *s
     }
   }
 
+  if (args->index_all) {
+    // Validate the arg (if it's not ENABLE or DISABLE -> throw an error)
+    if (!strcasecmp(args->index_all, "enable")) {
+      rule->index_all = true;
+    } else if (!strcasecmp(args->index_all, "disable")) {
+      rule->index_all = false;
+    } else {
+      QueryError_SetError(status, QUERY_EADDARGS, "Invalid argument for `INDEXALL`");
```

Comment:
maybe specify in the error message the valid arguments?

Reply:
In general, we should avoid doing so. User input may be binary data/breaking RESP and an error reply must be a simple string.
IMO saying which `KEY` has an invalid value is suggestive enough

Reply:
@GuyAv46 Not sure I follow, what would be the problem with having the following error: `Invalid argument for \`INDEXALL\`, use \`ENABLE/DISABLE\``?

Reply:
This is good as well. Including the supplied value might be bad (`Invalid argument for INDEXALL: 'banana'`)